examples/gsm8k/gsm8k.py

```diff
@@ -0,0 +1,350 @@
+                        f"Could not parse step answer: {text}. Returning None."
+                    )
+                new_state = state.copy()
+                new_state["current"] = int_answer
+                new_state["phase"] = 2
+                new_states.append(new_state)
+            else:
+                raise ValueError(f"Unknown method: {state['method']}")
+        return new_states
+    
+    def parse_aggregation_answer(self, response: str, **kwargs) -> Union[Dict, List[Dict]]:
+        pass
+
+    def parse_improve_answer(self, response: str, **kwargs) -> Dict:
+        pass
+
+    def parse_validation_answer(self, response: str, **kwargs) -> bool:
+        pass
+
+    def parse_score_answer(self, states: List[Dict], texts: List[str]) -> List[float]:
+        assert len(states) == 1, "Scoring multiple states is not implemented."
+        score = []
+        for text in texts:
+            if "True" in text and not "False" in text:
+                score.append(1.0)
+            else:
+                score.append(0.0)
+        return score
+
+
+def io() -> operations.GraphOfOperations:
+    """
+    Generates the Graph of Operations for the IO method.
+
+    :return: Graph of Operations
+    :rtype: GraphOfOperations
+    """
+    operations_graph = operations.GraphOfOperations()
+
+    operations_graph.append_operation(operations.Generate(1, 1))
+    operations_graph.append_operation(operations.Score(1, False))
+    operations_graph.append_operation(operations.GroundTruth(utils.test_answer))
+
+    return operations_graph
+
+def cot() -> operations.GraphOfOperations:
+    return io()
+
+def cotsc() -> operations.GraphOfOperations:
+    """
+    Generates the Graph of Operations for the CoT method with Self-Consistency.
+
+    :return: Graph of Operations
+    :rtype: GraphOfOperations
+    """
+
+    def score_by_majority_vote(states: List[Dict]) -> List[float]:
+        answers = [state["current"] for state in states]
+        answer_counts = Counter(answers)
+        most_common_answer, _ = answer_counts.most_common(1)[0]
+        scores = [1 if answer == most_common_answer else 0 for answer in answers]
+        return scores
+
+
+    operations_graph = operations.GraphOfOperations()
+
+    operations_graph.append_operation(operations.Generate(1, 5))
+    operations_graph.append_operation(operations.Score(combined_scoring=True, scoring_function=score_by_majority_vote))
+    operations_graph.append_operation(operations.KeepBestN(1, True))
+    operations_graph.append_operation(operations.Score(1, False))
+    operations_graph.append_operation(operations.GroundTruth(utils.test_answer))
+
+    return operations_graph
+
+def run(
+        data_ids: List[int],
+        methods: List[Callable[[], operations.GraphOfOperations]],
+        budget: float,
+        lm_name: str,
+    ) -> float:
+
+    orig_budget = budget
+    data_path = Path.cwd() / "datasets" / "grade-school-math" / "grade_school_math" / "data" / "test.jsonl"
```

Comment:
Again a next iteration thing, we could use the common utils code for the results code which is repeated in all the modules